pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go

```diff
@@ -103,49 +161,48 @@ func startDistIngestion(
 		)
 		defer recv.Release()
 
-		jobsprofiler.StorePlanDiagram(ctx, execCtx.ExecCfg().DistSQLSrv.Stopper, p, execCtx.ExecCfg().InternalDB,
-			ingestionJob.ID())
-
 		// Copy the evalCtx, as dsp.Run() might change it.
 		evalCtxCopy := *execCtx.ExtendedEvalContext()
-		dsp.Run(ctx, planCtx, noTxn, p, recv, &evalCtxCopy, nil /* finishedSetupFn */)
+		dsp.Run(ctx, planCtx, noTxn, initialPlan, recv, &evalCtxCopy, nil /* finishedSetupFn */)
 		return rw.Err()
 	}
 
 	updateRunningStatus(ctx, execCtx, ingestionJob, jobspb.Replicating,
 		"running the SQL flow for the stream ingestion job")
+	if err = ctxgroup.GoAndWait(ctx, execInitialPlan, replanner); errors.Is(err, sql.ErrPlanChanged) {
```

Comment:
When the error isn't defined in the iff block, I find it a bit clearer to use the more typical style:

```suggestion
	err = ctxgroup.GoAndWait(ctx, execInitialPlan, replanner)
	if errors.Is(err, sql.ErrPlanChanged) {
```